### ISSUE TITLE
owncloud integration issue

### ISSUE THREAD

**Issue by sonanchenko** [jsxc/jsxc#152]
Dear Colleagues,

Preconditions:
Fresh install, all services are on the same server
Owncloud 8.0.2
Openfire 3.9.3
JSXC 1.1.0

Openfire:
Enabled - Clients can connect to this server using HTTP binding. 
Port: 7070
SSL port: 7443
Script Syntax: Allows BOSH clients with limited access to connect to the server 

Owncloud JXSC Plugin settings:
BOSH url:  /http-bind
XMPP domain: compservice.in.ua

Owncloud's admin panel says that:
BOSH Server reachable.
 
Apache config:
LoadModule proxy_module /usr/lib/apache2/modules/mod_proxy.so
LoadModule proxy_http_module /usr/lib/apache2/modules/mod_proxy_http.so

ProxyRequests Off
ProxyPass /http-bind http://localhost:7070/http-bind/
ProxyPassReverse /http-bind http://localhost:7070/http-bind/
ProxyPass /http-binds http://localhost:7443/http-bind/
ProxyPassReverse /http-binds http://localhost:7443/http-bind/

I checked from server's console:
`> curl -d "<body rid='123456' xmlns:xmpp='urn:xmpp:xbosh' />" http://localhost/http-bind/
<body xmlns="http://jabber.org/protocol/httpbind" xmlns:stream="http://etherx.jabber.org/streams" from="compservice.in.ua" authid="?????" sid="?????" secure="true" requests="2" inactivity="30" polling="5" wait="60"><stream:features><mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl"><mechanism>DIGEST-MD5</mechanism><mechanism>PLAIN</mechanism><mechanism>CRAM-MD5</mechanism></mechanisms><compression xmlns="http://jabber.org/features/compress"><method>zlib</method></compression><bind xmlns="urn:ietf:params:xml:ns:xmpp-bind"/><session xmlns="urn:ietf:params:xml:ns:xmpp-session"/></stream:features></body>`

Issue description:
I am not able to connect to chat from owncloud.
I have not found any errors in owncloud.log
I checked javascript console:

no element found http-bind:1:1 (when I click in Firebug console - I see:
Source of https://compservice.in.ua/httpd-bind
```
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 400 Bad Request</title>
</head>
<body>
<h2>HTTP ERROR: 400</h2>
<p>Problem accessing /http-bind/. Reason:
<pre>    Bad Request</pre></p>
<hr /><i><small>Powered by Jetty://</small></i>
```
)

After that in console there is next error:
TypeError: elem.getAttribute is not a function jsxc.dep.min.js:27:8655

P.S.
I successfully connects from Pidgin with following settings:
Domain: compservice.in.ua
Resource: Work
Connection security:  Require encryption
Port: 5222
Bosh url: empty

So could you please clarify what I am doing wrong ?
I very appreciate your help.

Thank you in advance.

**Comment by v1r0x** [comment]
I don't know if this is related to your issue, but I think you have a missing `https` in your SSL Proxy:
ProxyPass /http-binds http __s__://localhost:7443/http-bind/
ProxyPassReverse /http-binds http __s__://localhost:7443/http-bind/

and is this a typo or where does the http __d__ come from?
> no element found http-bind:1:1 (when I click in Firebug console - I see:
> Source of https://compservice.in.ua/httpd-bind

**Comment by sonanchenko** [comment]
>and is this a typo or where does the http d come from?

you are right. just typo
real one: https://compservice.in.ua/http-bind

Also I have tried your suggestion. However still having issues:
new 2 errors are added to javascript console:
```
"NetworkError: 500 Task org.jivesoftwar...0, completed tasks = 0] - https://compservice.in.ua/http-bind"

<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 500 Task org.jivesoftware.openfire.http.HttpSessionManager$HttpPacketSender@1bf99b8 rejected
 from java.util.concurrent.ThreadPoolExecutor@1baff8a[Terminated, pool size = 0, active threads = 0,
 queued tasks = 0, completed tasks = 0]</title>
</head>
<body>
<h2>HTTP ERROR: 500</h2>
<p>Problem accessing /http-bind/. Reason:
<pre>    Task org.jivesoftware.openfire.http.HttpSessionManager$HttpPacketSender@1bf99b8 rejected from
 java.util.concurrent.ThreadPoolExecutor@1baff8a[Terminated, pool size = 0, active threads = 0, queued
 tasks = 0, completed tasks = 0]</pre></p>
<hr /><i><small>Powered by Jetty://</small></i>
```

https://compservice.in.ua/http-bind
returns
```
HTTP ERROR: 400

Problem accessing /http-bind/. Reason:

    Bad Request
Powered by Jetty://
```

**Comment by v1r0x** [comment]
you mix up your https and http proxy

if you use http __s__,  you have to use /http-bind __s__
if you use http, you have to use /http-bind

**Comment by sualko** [comment]
I would suggest to use only the non-ssl proxy, because this a inner-server connection and otherwise you could have problems with your bosh certificate.

```
ProxyPass /http-bind/ http://localhost:7070/http-bind/
ProxyPassReverse /http-bind/ http://localhost:7070/http-bind/
```

This is probably a server configuration issue, so please try to connect with pidgin via bosh. Otherwise send me some test credentials for further investigation.

**Comment by sualko** [comment]
Can we close this?

**Comment by sonanchenko** [comment]
Sorry for the late response - too much holidays in May ;)
I'll test your suggestions today or tomorrow and will back with feedback.

**Comment by sonanchenko** [comment]
Dear Colleagues,

It is a kind of magic.
I re-installed jsxc 2.0 from scratch 
hm... I also updated owncloud to 8.0.3 and openfire to 3.10.0

I successfully connected with the same parameters ;)
/http-bind as BOSH url and same redirects 
ProxyPass /http-bind/ http://localhost:7070/http-bind/
ProxyPassReverse /http-bind/ http://localhost:7070/http-bind/

So thanks to all.
I think, this case can be closed.
